toqito/channel_props/is_extremal.py

```diff
@@ -0,0 +1,64 @@
+"""Determines whether a quantum channel is extremal."""
+
+import numpy as np
+from numpy.linalg import matrix_rank
+
+from toqito.channel_ops.choi_to_kraus import choi_to_kraus
+
+
+def is_extremal(channel):
+    r"""Determine whether a quantum channel is extremal.
+
+    A channel with Kraus operators :math:`\{A_i\}_{i=1}^{r}` is **extremal** if and only if
+    the set :math:`\{A_i^\dagger A_j : i,j=1,\dots,r\}` is linearly independent.
+
+    For more details, see Section 2.2.4 of :cite:`Watrous_2018_TQI`.
+
+    :param channel: The quantum channel representation, which can be:
+        - A list of Kraus operators,
+        - A Choi matrix,
+        - A dictionary with ``'kraus'`` or ``'choi'`` key,
+        - An object with a callable ``kraus()`` method.
+    :type channel: list[numpy.ndarray] | numpy.ndarray | dict
+    :raises ValueError: If no Kraus operators can be extracted.
+    :return: ``True`` if the channel is extremal; ``False`` otherwise.
+    :rtype: bool
+    """
+    # Convert the input into a list of Kraus operators
+    if isinstance(channel, list):
+        kraus_ops = channel
+    elif isinstance(channel, np.ndarray):
+        kraus_ops = choi_to_kraus(channel)
+    elif isinstance(channel, dict):
+        if "kraus" in channel:
+            kraus_ops = channel["kraus"]
+        elif "choi" in channel:
+            kraus_ops = choi_to_kraus(channel["choi"])
+        else:
+            raise ValueError("Dictionary must have a 'kraus' or 'choi' key.")
+    elif hasattr(channel, "kraus") and callable(channel.kraus):
+        kraus_ops = channel.kraus()
```

Comment:
I don't see a scenario in which this would ever work as the input variable won't have this method associated to it.